### PR TITLE
Window Semaphore: Alignment!

### PR DESCRIPTION
**Pull request by jleandroperez**
### Description
In this PR we're adjusting the main Window's Semaphore position (Maximize / Minimize / Close buttons) so that they align perfectly with regards of other UI elements.

@eshurakov Sending you over a REALLY cool PR!! Thank youuu!!

cc @SylvesterWilmott Looks suuuupeeeeeeeeeerrrrrr

Ref. #718

### Details
- **New MainWindowController**: Encapsulates the Semaphore workaround
- **Main Window** is no longer initialized via nib
- Main Interface initialization mechanism was also updated

### Test: Semaphore Alignment!
1. Launch Simplenote

- [x] Verify the Semaphore's vertical position is centered with regards to the Note List's Title
- [x] Verify that if you hide the Tags List (CMD + T) the Semaphore looks great (and aligned with regards to the List Title)
- [ ] Verify that pressing **Shift + CMD + F** gets you into Focus Mode, and the Semaphore is aligned with regards to the Sidebar Toolbar Button
- [ ] Verify that you can drag the Window from anywhere in the Toolbar area (please!!) 😄 

### Test: Fullscreen
1. Launch Simplenote
2. Click over the Window's Fullscreen button

- [x] Verify that the top area looks great (no flickers?)
- [x] Verify that moving the mouse upwards reveals the System Bar, and the Semaphore looks vertically centered in its container
- [x] Verify that exiting Fullscreen mode causes the main Window to look great (and the semaphore lights still properly aligned)

### Test: Autosave
1. Launch the app
2. Relocate / resize the main window
3. Press CMD + Q to quit
4. Relaunch Simplenote

- [x] Verify the last Window Frame is properly restored

### Release
These changes do not require release notes.

### Screenshots

Before|Now
-|-
Old|New

### CODE REVIEW

**Comment by jleandroperez** [comment]
Oh!!! I almost forget!! **Important!!!**

Thank you @emilylaguna 🔥 !!

File: Simplenote/Base.lproj/MainMenu.xib [hunk 1]
```diff
@@ -3,7 +3,6 @@
     <dependencies>
         <deployment identifier="macosx"/>
         <plugIn identifier="com.apple.InterfaceBuilder.CocoaPlugin" version="17506"/>
-        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
```
Comment (eshurakov) [inline Simplenote/Base.lproj/MainMenu.xib:6]: Bye bye Xcode 8 👋

**Review by eshurakov — APPROVED** [review]
Looks nice!

1. I tried dragging the window by the toolbar. Horizontally it worked everywhere, but vertically it works only in the top half. In the bottom half the cursor turns from the pointer into caret.

2. I noticed that the semaphore is not exactly aligned with sidebar button:
Screenshot 2020-12-09 at 20 39 55

3. And also it feels that left side is not aligned with anything :-)
Screenshot 2020-12-09 at 20 31 54

**Comment by jleandroperez** [comment]
Tracking issues (1) and (2) in #718. As per (3): Alignment, I'm afraid that would be by design.

Thanks a looot Evgeny!!!

**Pull request merged**